core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -409,47 +409,42 @@ public static function loadLanguageFile($strName, $strLanguage=null, $blnNoCache
 		// Fall back to English
 		$arrCreateLangs = ($strLanguage == 'en') ? array('en') : array('en', $strLanguage);
 
-		// Prepare the XLIFF loader
-		$xlfLoader = new XliffFileLoader(static::getContainer()->getParameter('kernel.project_dir'), true);
-
 		$strCacheDir = static::getContainer()->getParameter('kernel.cache_dir');
+		$resLoaderDelegator = static::getContainer()->get('contao.resource_loader.delegator');
 
 		// Load the language(s)
 		foreach ($arrCreateLangs as $strCreateLang)
 		{
+			$cachedFile = $strCacheDir . '/contao/languages/' . $strCreateLang . '/' . $strName . '.php';
+
 			// Try to load from cache
-			if (file_exists($strCacheDir . '/contao/languages/' . $strCreateLang . '/' . $strName . '.php'))
-			{
-				include $strCacheDir . '/contao/languages/' . $strCreateLang . '/' . $strName . '.php';
-			}
-			else
+			if (file_exists($cachedFile))
 			{
 				try
 				{
-					$files = static::getContainer()->get('contao.resource_locator')->locate('languages/' . $strCreateLang . '/' . $strName . '.php', null, false);
-				}
-				catch (\InvalidArgumentException $e)
-				{
-					$files = array();
+					$resLoaderDelegator->load($cachedFile);
+					continue;
 				}
+				catch (\Exception $e) {}
+			}
 
-				foreach ($files as $file)
-				{
-					include $file;
-				}
+			// Find the given filename either as .php or .xlf file (Contao supports both)
+			$finder = static::getContainer()->get('contao.resource_finder')->findIn('languages/' . $strCreateLang)->name('/^' . $strName . '\.(php|xlf)$/');
 
+			/** @var SplFileInfo $file */
+			foreach ($finder as $file)
+			{
 				try
 				{
-					$files = static::getContainer()->get('contao.resource_locator')->locate('languages/' . $strCreateLang . '/' . $strName . '.xlf', null, false);
+					$resLoaderDelegator->load($file->getRealPath(), $strCreateLang);
 				}
-				catch (\InvalidArgumentException $e)
+				catch (\Exception $e)
 				{
-					$files = array();
-				}
-
-				foreach ($files as $file)
-				{
-					$xlfLoader->load($file, $strCreateLang);
+					static::getContainer()->get('monolog.logger.contao')->log(
```

Comment:
@mvscheidt Now we are logging an error if the requested file does not exist. In Contao 4.6 we have just silently ignored this. Besides that this is a behavior change, what is the point of logging the error? Should we not not catch the exception instead in this case?

@contao/developers Should we keep silently ignoring non existing language files?

Reply:
> Besides that this is a behavior change, what is the point of logging the error?

Making it a bit easier to debug in case some language file cannot be loaded. Before you would have to go through the Contao Code to get a "first idea" what could be going wrong if it wasn't completely obvious.

That said it is mostly personal preference because I dislike wading through code only to find a silently catched Exception which would've literally told me what's wrong in seconds if it was logged. Not in this case specifically, but maybe it helps other people who are not developers specifically - they still need to interpret the log, but they don't need to read through the code (which is harder than trying to understand something from the log I'd say).

> Should we not not catch the exception instead in this case?

Wouldn't that cause an error in the Frontend / Backend then (depending where the error is caused)?

From a users perspective having an update "break" the site might not be desired unless it's a major version change (where this is to be expected), or am I seeing that wrong?

For example say someone has installed an extension which causes an error (albeit silently) there - now it would cause an error in the frontend / backend, wouldn't it?

Reply:
Actually the case can never occur I think, because the resource finder only returns existing files and the loader only throws an exception if a file is missing. 😄 

> From a users perspective having an update "break" the site might not be desired

I agree, although that depends on how you define "break". If the exception is caught, the translations will not be loaded and the user might end up with English labels on their German page. Isn't that already broken?

Reply:
Ok, I agree that we should log exceptions. Probably not when including the cache file and the `langconfig.php` file, because both are PHP files, which cannot be broken (otherwise they would not have been created at all) and which are only included (so there no exceptions), but in all other cases.